### ISSUE TITLE
FR: Use the Moodle activity icons inside the collapsible activity node instead of the generic "activity" icon

### ISSUE THREAD

**Issue by danowar2k** [moodleuulm/moodle-local_boostnavigation#65]
It was mentioned to us that the collapsible activity node only displays activities with the generic "activity" icon (originally a database icon, FontAwesome mapped to fa-share-alt).

It would be nice if the activity nodes would have the Moodle activity icons. Is this something that would be worthy of a pull request? If so, would you like a replacement or an additional setting (boolean "Use Moodle core icons for activities" or something like that)?

**Comment by abias** [comment]
Hi Daniel,

thank you for this nice improvement proposal. In fact, I wanted to realize the same when I added the activity nodes to the plugin, but, back then, it was more time-consuming than initially expected and I had to postpone it.

Here's why:
* The icons in the navigation nodes are, at the last stage of processing and when they are shown in the browser, FontAwesome icons. However, when the navigation nodes are created in PHP, the icons are set as Moodle pix icons. This gap is bridged with the local_boostnavigation_get_fontawesome_icon_map() function in lib.php which is a callback function with which each plugin can add its own FA mappings to the rendered theme. This concept was made for backwards compatibility when FontAwesome was added to Boost. 
* FontAwesome does not know anything about Moodle activity icons, you won't find them in the FontAwesome iconset. (In fact, FontAwesome does not even know anything about Moodle. There is a Moodle logo icon request pending for a long time on https://github.com/FortAwesome/Font-Awesome/issues/2167.)
* If there is no FA mapping for a particular pix icon, Moodle looks for a fallback icon. First for a SVG and then for a PNG file. These files are also used by themes which do not know anything about FontAwesome. That's why we have these fallback files in https://github.com/moodleuulm/moodle-local_boostnavigation/tree/master/pix.
* There is a SVG and a PNG icon for each activity plugin, the SVG icon for mod_assign for example is on https://github.com/moodle/moodle/blob/master/mod/assign/pix/icon.svg. 
* So it wouldn't be a problem at all to add the activity icons as pix icons to each activity navigation node. They would be displayed as SVG or PNG, without any need for a FontAwesome icons. But: The activity icons are colored, the navigation drawer is monochromatic!
* To align the activity icons with the other nav drawer icons, you would have to a) make them monochromatic and b) match their darkness to the dark grey tone of the FA icons. You can do this manually by forking the icons into this plugin which will turn out to be a never-ending task as soon as new activities are expected to work with this feature. Or you can do this at runtime in the browser with SVG filters (https://caniuse.com/#feat=svg-filters) which should work but which I have never tried before.

Summing up:
Feel free to build a solution, I would be grateful to review your PR.

**Comment by abias** [comment]
Hi @danowar2k ,

the possibility to change the icons is now implemented in https://github.com/moodleuulm/moodle-local_boostnavigation/commit/d4318caa850a9d5e56ce2d86f7712dd13345e9c3. I also tried to greyscale the icons with CSS, but that didn't look good.

I hope that the patch helps you anyway.

Cheers,
Alex

**Comment by danowar2k** [comment]
Thank you very much for your work! (sorry I couldn't focus on a PR myself)
